packages/vite-plugin/src/index.ts

```diff
@@ -71,32 +99,38 @@ export function vanillaExtractPlugin({
         return null;
       }
 
-      const usedIndex = id.indexOf('?used');
-      const fixedId = usedIndex > 0 ? id.substring(0, usedIndex) : id;
-
-      if (ssr || useRuntime) {
+      if (ssr) {
+        const validId = id.substring(0, id.indexOf('?'));
         return addFileScope({
           source: code,
-          filePath: normalizePath(path.relative(packageInfo.dirname, fixedId)),
+          filePath: normalizePath(path.relative(packageInfo.dirname, validId)),
           packageInfo,
         }).source;
       }
 
       const { source, watchFiles } = await compile({
-        filePath: fixedId,
+        filePath: id,
         cwd: config.root,
       });
 
       for (const file of watchFiles) {
-        this.addWatchFile(file);
+        if (config.command === 'build' || file !== id) {
```

Comment:
Might be worth adding a comment as to why we have the `config.command === 'build'` check.